src/Text/Pandoc/Readers/Textile.hs

```diff
@@ -246,26 +258,47 @@ orderedListItemAtDepth = genericListItemAtDepth '#'
 -- | Common implementation of list items
 genericListItemAtDepth :: PandocMonad m => Char -> Int -> TextileParser m Blocks
 genericListItemAtDepth c depth = try $ do
-  count depth (char c) >> attributes >> whitespace
+  genericListItemStartAtDepth c depth
+  genericListItemContentsAtDepth depth
+
+-- | Lists always start with a number of leading characters '#' or '*'
+-- Ordered list start characters '#' can be followed by the start attribute
+-- number, but bullet list characters '*' can not
+genericListStartAtDepth :: PandocMonad m => Char -> Int -> TextileParser m Int
```

Comment:
I also think it is a good idea to replace `genericListStartAtDepth` with two separate parsers for Ordered and Unordered lists, since the syntax for them are significantly different now. I'll be happy to open a separate "cleanup" PR to address this

Reply:
Squeezing that in here would be fine, but a separate PR would work, too.

Reply:
Changes squeezed in accordingly. Thank you!